buildSrc/src/main/java/org/elasticsearch/gradle/ReaperPlugin.java

```diff
@@ -27,14 +28,21 @@ public void apply(Project project) {
 
         project.getPlugins().apply(GlobalBuildInfoPlugin.class);
 
-        Path inputDir = project.getRootDir()
+        File inputDir = project.getRootDir()
             .toPath()
             .resolve(".gradle")
             .resolve("reaper")
-            .resolve("build-" + ProcessHandle.current().pid());
-        ReaperService service = project.getExtensions()
-            .create("reaper", ReaperService.class, project, project.getBuildDir().toPath(), inputDir);
+            .resolve("build-" + ProcessHandle.current().pid())
+            .toFile();
 
-        project.getGradle().buildFinished(result -> service.shutdown());
+        var reaperServiceProvider = project.getGradle().getSharedServices().registerIfAbsent("reaper", ReaperService.class, spec -> {
+            // Provide some parameters
+            spec.getParameters().getInputDir().set(inputDir);
+            spec.getParameters().getBuildDir().set(project.getBuildDir());
```

Comment:
Would the more correct thing be to use `project.getLayout().getBuildDirectory()` here so it's providers all the way through?

Reply:
yes

Reply:
some clarification. It's better to use an injected ProjectLayout instance here (which I changed here to do so). This way we don't rely on the mutable `Project` object at all.